presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestAddNotNullFiltersToJoinNode.java

```diff
@@ -0,0 +1,327 @@
+    }
+
+    @Test
+    public void testNotNullPredicatesAddedOnlyForInnerSideTablesVariableReferences()
+    {
+        String query = "select 1 from lineitem l left join orders o on l.orderkey = o.orderkey and partkey - custkey > 10";
+        assertPlan(query,
+                anyTree(
+                        join(LEFT,
+                                ImmutableList.of(equiJoinClause("LINE_ORDER_KEY", "ORDERS_ORDER_KEY")),
+                                Optional.of("partkey - custkey > 10"),
+                                anyTree(
+                                        tableScan("lineitem", ImmutableMap.of(
+                                                "LINE_ORDER_KEY", "orderkey",
+                                                "partkey", "partkey"))),
+                                anyTree(
+                                        filter("ORDERS_ORDER_KEY IS NOT NULL and custkey IS NOT NULL",
+                                                tableScan("orders",
+                                                        ImmutableMap.of(
+                                                                "ORDERS_ORDER_KEY", "orderkey",
+                                                                "custkey", "custkey")))))));
+
+        query = "select 1 from lineitem l right join orders o on l.orderkey = o.orderkey and custkey > partkey";
+        assertPlan(query,
+                anyTree(
+                        join(RIGHT,
+                                ImmutableList.of(equiJoinClause("LINE_ORDER_KEY", "ORDERS_ORDER_KEY")),
+                                Optional.of("custkey > partkey"),
+                                anyTree(
+                                        filter("LINE_ORDER_KEY IS NOT NULL and partkey IS NOT NULL",
+                                                tableScan("lineitem", ImmutableMap.of(
+                                                        "LINE_ORDER_KEY", "orderkey",
+                                                        "partkey", "partkey")))),
+                                anyTree(
+                                        tableScan("orders",
+                                                ImmutableMap.of(
+                                                        "ORDERS_ORDER_KEY", "orderkey",
+                                                        "custkey", "custkey"))))));
+    }
+
+    @Test(dataProvider = "equiJoinTestCases")
+    public void testNotNullInferenceForEquiJoinOnlyStrategy(String filterSql, String[] expectedInferredNotNullVariables)
+    {
+        assertInferredNotNullVariableRefsListMatch(EQUI_JOIN_ONLY, filterSql, buildVariableReferencesList(expectedInferredNotNullVariables));
+    }
+
+    @Test(dataProvider = "standardOperatorTestCases")
+    public void testNotNullInferenceForInferFromStandardOperatorsStrategy(String filterSql, String[] expectedInferredNotNullVariables)
```

Comment:
can you add negative tests too (nonStandardOperatorTestCases does not infer nulls)

Reply:
[Added here](https://github.com/prestodb/presto/blob/82ec0a08d8533a98dc8ef371721db63165f10821/presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestAddNotNullFiltersToJoinNode.java#L123-L127)